### PR TITLE
test: add tests for `IMPORT INTO` statement with Foreign Key

### LINKED ISSUE #56563: Add test for `IMPORT INTO` with `Foreign Key`.
## Enhancement

It's expected that the `IMPORT INTO` will ignore foreign key check. We need to add a test to keep the behavior.

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Issue Number: close #56563

### What changed and how does it work?

Add tests for using both `IMPORT INTO` statement and the foreign key. It's expected that `IMPORT INTO` will always ignore the foreign key checks.

TODO: confirm that the behavior is as expected, and modify the document.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/importintotest/import_into_test.go [hunk 1]
```diff
@@ -1239,3 +1239,26 @@ func (s *mockGCSSuite) TestBadCases() {
...
+	filePath := path.Join(tempDir, "file.csv")
+	s.NoError(os.WriteFile(filePath, content, 0644))
+	s.prepareAndUseDB("import_into")
+	s.tk.MustExec("create table parent (id int primary key);")
+	s.tk.MustExec("create table child (id int primary key, fk int, foreign key (fk) references parent(id));")
+	sql := fmt.Sprintf(`IMPORT INTO import_into.child
+		FROM 'gs://foreign-key-test/child.csv?endpoint=%s'`, gcsEndpoint)
+
+	// it should success even if the parent table is empty
+	s.tk.MustQuery(sql)
```
Comment (lance6716) [inline tests/realtikvtest/importintotest/import_into_test.go:1263]: maybe also check SELECT result

**Review by lance6716 — APPROVED** [review]

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 1 above)
Comment (YangKeao) [reply tests/realtikvtest/importintotest/import_into_test.go:1263]: Done.

File: tests/realtikvtest/importintotest/import_into_test.go [hunk 2]
```diff
@@ -1239,3 +1239,26 @@ func (s *mockGCSSuite) TestBadCases() {
...
+	s.server.CreateObject(fakestorage.Object{
+		ObjectAttrs: fakestorage.ObjectAttrs{
+			BucketName: "foreign-key-test",
+			Name:       "child.csv",
+		},
+		Content: content,
+	})
+	tempDir := s.T().TempDir()
+	filePath := path.Join(tempDir, "file.csv")
+	s.NoError(os.WriteFile(filePath, content, 0644))
```
Comment (D3Hunter) [inline tests/realtikvtest/importintotest/import_into_test.go:1255]: why write to local, seems not used

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 1 above)
Comment (D3Hunter) [reply tests/realtikvtest/importintotest/import_into_test.go:1263]: have you pushed code, i didn't see it

**Review by D3Hunter — APPROVED** [review]

**Comment by D3Hunter** [comment]
/hold

please unhold after fix comments

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 2 above)
Comment (YangKeao) [reply tests/realtikvtest/importintotest/import_into_test.go]: Nice catch. It's not used and has been removed.

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 1 above)
Comment (YangKeao) [reply tests/realtikvtest/importintotest/import_into_test.go:1260]: > have you pushed code, i didn't see it

Oops. Network issue :facepalm:. Pushed now.

**Pull request merged**